apps/console/app/mocks/branches.ts

```
@@ -31,11 +33,63 @@ export const branchHandlers = [
 
   http.get<{ agentSlug: string }>(
     "/api/v1/agents/:agentSlug/branches",
-    async () => {
-      const branches = db.branch.getAll();
+    async ({ params }) => {
+      try {
+        const branches = db.branch.findMany({
+          where: { agentSlug: { equals: params.agentSlug } },
+        });
+
+        await delay(1000);
+        return HttpResponse.json(branches);
+      } catch {
+        return new HttpResponse("Failed to fetch branches", { status: 500 });
+      }
+    },
+  ),
 
-      await delay(1000);
-      return HttpResponse.json(branches);
+  // Update entire branch (including models JSON object)
+  http.put<{ agentSlug: string; branchSlug: string }>(
+    "/api/v1/agents/:agentSlug/branches/:branchSlug",
+    async ({ request, params }) => {
+      const body = (await request.json()) as {
+        name?: string;
+        models?: Array<{ alias: string; type: string; endpoint?: any }>;
+      };
+
+      try {
+        const branch = db.branch.findFirst({
+          where: {
+            agentSlug: { equals: params.agentSlug },
+            slug: { equals: params.branchSlug },
+          },
+        });
+
+        if (!branch) {
+          return new HttpResponse("Branch not found", { status: 404 });
+        }
+
+        // Update branch properties
+        if (body.name) {
+          branch.name = body.name;
+          branch.slug = slugify(body.name, { lower: true, strict: true });
+        }
+
+        // Update models JSON object
+        if (body.models) {
+          branch.models = body.models;
+        }
+
+        // Persist the changes to the database
+        db.branch.update({
+          where: { id: { equals: branch.id } },
+          data: branch,
+        });
+
+        await delay(500);
+        return HttpResponse.json(branch);
+      } catch {
+        return new HttpResponse("Failed to update branch", { status: 500 });
```

Why do you try/catch to still return a 500? Do you think this is necessary? Why not just return the original error?

---

I will return the original error like this
```ts
return new HttpResponse(error instanceof Error ? error.message : String(error), { status: 500 },);
```